pkg/profile/artifact.go

```diff
@@ -88,22 +91,30 @@ func (p *Profile) makeArtifacts(profileRepos []string, gitClient git.Git) ([]Art
 			if err != nil {
 				return nil, fmt.Errorf("failed to generate resources for nested profile %q: %w", artifact.Name, err)
 			}
-			for i := range nestedArtifacts {
-				nestedArtifacts[i].Name = filepath.Join(artifact.Name, nestedArtifacts[i].Name)
-			}
 			artifacts = append(artifacts, nestedArtifacts...)
+			p.nestedName = ""
 		case profilesv1.HelmChartKind:
-			a.Objects = append(a.Objects, p.makeHelmRelease(artifact, profileRepoPath))
+			var o runtime.Object
+			helmRelease := p.makeHelmRelease(artifact, profileRepoPath)
 			if artifact.Path != "" {
-				a.Objects = append(a.Objects, p.makeGitRepository())
+				if p.gitRepositoryNamespace == "" && p.gitRepositoryName == "" {
+					return nil, fmt.Errorf("in case of local resources, the flux gitrepository object's details must be provided")
+				}
+				helmRelease.Spec.Chart.Spec.Chart = filepath.Join(p.rootDir, "artifacts", artifact.Name, artifact.Path)
+				o = p.makeGitRepository(artifact.Path)
 			}
 			if artifact.Chart != nil {
-				a.Objects = append(a.Objects, p.makeHelmRepository(artifact.Chart.URL, artifact.Chart.Name))
+				o = p.makeHelmRepository(artifact.Chart.URL, artifact.Chart.Name)
 			}
+			a.Objects = append(a.Objects, helmRelease, o)
 			artifacts = append(artifacts, a)
 		case profilesv1.KustomizeKind:
-			a.Objects = append(a.Objects, p.makeKustomization(artifact, profileRepoPath))
-			a.Objects = append(a.Objects, p.makeGitRepository())
+			if p.gitRepositoryNamespace == "" && p.gitRepositoryName == "" {
+				return nil, fmt.Errorf("in case of local resources, the flux gitrepository object's details must be provided")
+			}
+			path := filepath.Join(p.rootDir, "artifacts", artifact.Name, artifact.Path)
+			a.Objects = append(a.Objects, p.makeKustomization(artifact, path))
+			a.Objects = append(a.Objects, p.makeGitRepository(artifact.Path))
```

Comment:
could we split all of the contents of each `case` into a `func GenerateXArtifact` or somthing? This was already too big :smile:

Reply:
I didn't want to do that because https://github.com/weaveworks/pctl/issues/115 proposes to use builders in any case.